algorithms/src/sorting/Kokkos_SortByKeyPublicAPI.hpp

```diff
@@ -0,0 +1,109 @@
+  using ValuesType = Kokkos::View<ValuesDataType, ValuesProperties...>;
+  ::Kokkos::Impl::static_assert_is_admissible_to_kokkos_sort_by_key(keys);
+  ::Kokkos::Impl::static_assert_is_admissible_to_kokkos_sort_by_key(values);
+
+  static_assert(SpaceAccessibility<ExecutionSpace,
+                                   typename KeysType::memory_space>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the keys View argument!");
+  static_assert(
+      SpaceAccessibility<ExecutionSpace,
+                         typename ValuesType::memory_space>::accessible,
+      "Kokkos::sort: execution space instance is not able to access "
+      "the memory space of the values View argument!");
+
+  static_assert(KeysType::static_extent(0) == 0 ||
+                ValuesType::static_extent(0) == 0 ||
+                KeysType::static_extent(0) == ValuesType::static_extent(0));
+  if (values.size() != keys.size())
+    Kokkos::abort("values and keys extents must be the same");
+
+  if (keys.extent(0) <= 1) {
+    return;
+  }
+
+  ::Kokkos::Impl::sort_by_key_device_view_without_comparator(exec, keys,
+                                                             values);
+}
+
+// ---------------------------------------------------------------
+// overloads supporting a custom comparator
+// ---------------------------------------------------------------
+
+template <class ExecutionSpace, class ComparatorType, class KeysDataType,
+          class... KeysProperties, class ValuesDataType,
+          class... ValuesProperties>
+void sort_by_key(
+    const ExecutionSpace& exec,
+    const Kokkos::View<KeysDataType, KeysProperties...>& keys,
+    const Kokkos::View<ValuesDataType, ValuesProperties...>& values,
+    const ComparatorType& comparator) {
+  // constraints
+  using KeysType   = Kokkos::View<KeysDataType, KeysProperties...>;
+  using ValuesType = Kokkos::View<ValuesDataType, ValuesProperties...>;
+  ::Kokkos::Impl::static_assert_is_admissible_to_kokkos_sort_by_key(keys);
+  ::Kokkos::Impl::static_assert_is_admissible_to_kokkos_sort_by_key(values);
+
+  static_assert(SpaceAccessibility<ExecutionSpace,
+                                   typename KeysType::memory_space>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the keys View argument!");
+  static_assert(
+      SpaceAccessibility<ExecutionSpace,
+                         typename ValuesType::memory_space>::accessible,
+      "Kokkos::sort: execution space instance is not able to access "
+      "the memory space of the values View argument!");
+
+  static_assert(KeysType::static_extent(0) == 0 ||
+                ValuesType::static_extent(0) == 0 ||
+                KeysType::static_extent(0) == ValuesType::static_extent(0));
+  if (values.size() != keys.size())
+    Kokkos::abort("values and keys extents must be the same");
```

Comment:
Why not inject the labels and actual extents to increase the quality of the error message?